### PR TITLE
chore: update docs on versioning

### PR DESCRIPTION
**Pull request by robbie-c**
## Changes
Following the instructions in the docs can get us in a weird state, so I changed them.

See internal thread https://posthog.slack.com/archives/C02GS3N743W/p1713272909176839?thread_ts=1713272170.819489&cid=C02GS3N743W

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers
- [x] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "posthog-js",
-    "version": "1.123.0",
+    "version": "1.123.1",
```
Comment (benjackwhite) [inline package.json:3]: Wait - is this on purpose?

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (robbie-c) [reply package.json:3]: Yeah, this commit has a tag attached to it, where I tried (and failed) to create a new release locally. Ideally should be merged.

Comment (robbie-c) [reply package.json:3]: Ah wait we only have squash commits for this repo. Ah well it doesn't really matter - this PR is a minor bump wheras that was a patch

Comment (robbie-c) [reply package.json]: have removed it

**Pull request merged**